src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const versionsOfState = [];
+
+  for (const object of actions) {
+    switch (object.type) {
+      case 'addProperties':
+        const { extraData } = object;
+
+        Object.assign(copyState, extraData);
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = object;
+
+        removeProperties(copyState, keysToRemove);
+        break;
+
+      case 'clear':
+        copyState = {};
+        break;
+
+      default:
+        break;
+    }
+
+    versionsOfState.push({ ...copyState });
+  }
+
+  return versionsOfState;
+
+  function removeProperties(object, propertiesToRemove) {
+    for (const property of propertiesToRemove) {
+      delete object[property];
+    }
+  }
```

Comment:
let's get rid of the nested function and move it after our `transformStateWithClones` function